src/main/java/com/vint/shop/domain/User.java

```diff
@@ -1,68 +1,92 @@
 package com.vint.shop.domain;
 
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
 import javax.persistence.*;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
 import java.time.LocalDate;
+import java.util.Collection;
 import java.util.Objects;
+import java.util.Set;
 
 @Entity
 @Table(name = "\"user\"")
-public class User { //изменить название класса
+public class User implements UserDetails {
     @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column(name = "id")
-    private int id;
-    @Column(name = "password_hash")
-    private String password_hash;
+    private long id;
+
+    @NotEmpty(message = "Login cannot be empty")
+    @Size(min = 5, max = 50, message = "Login must be between 5 and 50 characters\n")
+    @Column(name = "login")
+    private String login;
 
-    @Column(name = "first_name")
-    private String first_name;
+    @Column(name = "username")
+    private String username;
```

Comment:
You SHOULDN'T introduce new column `username`. You should use login as your username and don't change the structure of DB

Reply:
login to enter, username - full name for order